### PR TITLE
Fix form issues

### LINKED ISSUE #1406: Submitter submission form – `Description` not hidden when id is assigned
The `Description` field in step 3 should be hidden and auto-filled when a submitter assigns a report an `incident id` in step 1

image

image

### PR DESCRIPTION
**Pull request by lmcnulty**
Resolves #1406, #1407, #1429, #1430, #1479, and #1266 (superceding #1322).

### To-do
- [x] Add tests
- [x] Attempt breakages

### Breakage Attempts
- [x] Promote an old submission to an incident and verify that it's added to the database
- [x] Promote an old submission to an issue and verify that it's added to the database
- [x] Add an old submission to a current incident
- [x] Create a new submission without an incident and filling only the required fields.  Promote it to an **incident**.
  - **Found problem** (Non-blocking): The fields that are mandatory for an issue/new incident are not marked, and it's weird to have them under "Tell us more..."
- [x] Create a new submission without an incident and filling only the required fields.  Promote it to an **issue**.
- [x] Create a new submission without an incident ID, filling out every possible field. Add it as a new **incident**. 
- [x] Create a new submission without an incident ID, filling out every possible field. Add it as a new **issue**. 
  - **Found problem** (Non-blocking): You get a prompt warning that data not associated with issues will be deleted, but it doesn't say what data that is specifically.
- [x] Create a submission with an incident ID and only the required fields. Add it to an existing incident.
- [x] Create a submission with an incident ID and only the required fields. Add it as an issue.
- [x] Edit an existing submission
  - **Found problem** (Non-blocking): Submitting the form was blocked because the title was too short, but that was not visibly indicated in that form.
- [x] Try to submit a report with no content
- [x] Edit a report on /cite/edit
- [x] Remove mandatory field on /cite/edit
  - **Found problem** (Non-blocking): The form validation error outline only appears when the field is selected. That makes it hard to find the problem.
- [x] Edit an incident on /incidents/edit
  - **Found problem** (Non-blocking): Description input is only one line
    - [x] Fixed 
- [x] Remove Alleged Developer on /incidents/edit
  - **Found problem**: Allows removal of field that's mandatory to submit.
    - [x] Fixed
- [x] Submit a new incident report with a description of 500 characters and confirm that it doesn't get truncated.
  - **Found Problem** (Non-blocking): The description isn't visible on the submission review
---

I think the problems with the ingest form can be handled in a different PR. It seems like selecting whether the report should be added as an incident or an issue should be handled within the ingest form, so it can show the appropriate fields.

### CODE REVIEW

File: site/realm/functions/promoteSubmissionToReport.js [hunk 1]
```diff
@@ -22,11 +22,11 @@ exports = async (input) => {
       const lastIncident = await incidents.find({}).sort({incident_id: -1}).limit(1).next();
       
       const newIncident = {
-        title: submission.title,
+        title: submission.incident_title || submission.title,
```
Comment (cesarvarela) [inline site/realm/functions/promoteSubmissionToReport.js:25]: Please update the test associated with this file.

**Review by cesarvarela — CHANGES_REQUESTED** [review]

**Comment by cesarvarela** [comment]
@lmcnulty please merge staging.

File: site/gatsby-site/cypress/e2e/integration/incidents.cy.js [hunk 2]
```diff
@@ -38,6 +40,8 @@ describe('Incidents', () => {
 
     cy.contains('label', 'Editors').next().find('[type="text"]').type('Test Editor{enter}');
 
+    cy.wait(10000);
```
Comment (cesarvarela) [inline site/gatsby-site/cypress/e2e/integration/incidents.cy.js:43]: This wait should be instead a `timeout` in the `cy.get` that queries the toast below.

File: site/gatsby-site/cypress/e2e/integration/incidents.cy.js [hunk 3]
```diff
@@ -27,6 +27,8 @@ describe('Incidents', () => {
       date: '2021-01-02',
     };
 
+    cy.wait(10000);
```
Comment (cesarvarela) [inline site/gatsby-site/cypress/e2e/integration/incidents.cy.js:30]: Try using `cy.waitForStableDom` instead.

File: site/gatsby-site/src/components/incidents/IncidentForm.js [hunk 4]
```diff
@@ -25,9 +25,19 @@ export const schema = Yup.object().shape({
...
   description: Yup.string().required(),
   date: Yup.date().required(),
-  AllegedDeployerOfAISystem: Yup.array(),
-  AllegedDeveloperOfAISystem: Yup.array(),
-  AllegedHarmedOrNearlyHarmedParties: Yup.array(),
+  AllegedDeployerOfAISystem: Yup.array().required(),
+  AllegedDeveloperOfAISystem: Yup.array().required(),
+  AllegedHarmedOrNearlyHarmedParties: Yup.array().required(),
+  editors: Yup.string()
+    .matches(/^.{3,}$/, {
```
Comment (cesarvarela) [inline site/gatsby-site/src/components/incidents/IncidentForm.js:32]: No need to change it, but there is `min` and `max` available.

**Review by cesarvarela — CHANGES_REQUESTED** [review]

File: site/gatsby-site/cypress/e2e/integration/incidents.cy.js (hunk 3 above)
Comment (lmcnulty) [reply site/gatsby-site/cypress/e2e/integration/incidents.cy.js]: Whoops, those were just there so I could inspect what was going on. Removed.

**Comment by smcgregor** [comment]
@lmcnulty when you are back in, please ping me on this one so we can get it into production. @LKchemposer is blocked on a few items here.

**Comment by lmcnulty** [comment]
@smcgregor I think this one's ready as it is. Merging staging to make sure the tests still pass.

**Pull request merged**